crates/sui-graphql-rpc/src/types/checkpoint.rs

```diff
@@ -229,6 +232,39 @@ impl Checkpoint {
 
         Ok(conn)
     }
+
+    /// Queries the database for the available range supported by the graphql server. This method
+    /// takes a connection, so that it can be used in an execute_repeatable transaction.
+    pub(crate) fn available_range(
+        conn: &mut PgConnection,
+    ) -> Result<(u64, u64), diesel::result::Error> {
+        use checkpoints::dsl as checkpoints;
+        use objects_snapshot::dsl as snapshots;
+
+        let checkpoint_range: Vec<i64> = conn.results(move || {
+            let rhs = checkpoints::checkpoints
+                .select(checkpoints::sequence_number)
+                .order(checkpoints::sequence_number.desc())
+                .limit(1);
+
+            let lhs = snapshots::objects_snapshot
+                .select(snapshots::checkpoint_sequence_number)
+                .order(snapshots::checkpoint_sequence_number.desc())
+                .limit(1);
+
+            lhs.union(rhs)
+        })?;
+
+        Ok(match checkpoint_range.as_slice() {
+            [] => (0, 0),
+            [single_value] => (0, *single_value as u64),
+            values => {
+                let min_value = *values.iter().min().unwrap();
+                let max_value = *values.iter().max().unwrap();
```

Comment:
Ideally, we should not unwrap and instead take care of this error. Maybe something along the lines?
```suggestion
                let min_value = *values.iter().min().ok_or(Error::InternalError("Cannot get available range"))?;
```

Reply:
This code cannot fail, because we know that the range has at least two elements at this point.  A `// SAFETY` comment is worthwhile here, but no need to use `ok_or`.